libs/trainer/mlpe/trainer/trainer.py

```diff
@@ -222,8 +221,13 @@ def train(
     logging.info(flow)
     logging.info("Initializing loss and optimizer")
 
-    optimizer = optimizer_fn(flow.parameters(), lr=lr, **optimizer_kwargs)
```

Comment:
Let's not revert this bit

Reply:
I think we should for clarity (this is what we have in BBHnet). Given the optimizer is not our limiting factor, I think we can hold off on this until we have a better solution

Reply:
Yeah, unfortunately I think we'll have to do this by hand for now given `typeo` constraints

Reply:
I fail to understand what the constraint is.